sklearn/datasets/_openml.py

```diff
@@ -105,22 +106,25 @@ def is_gzip_encoded(_fsrc):
         return fsrc
 
     local_path = _get_local_path(openml_path, data_home)
+    dir_name, file_name = os.path.split(local_path)
     if not os.path.exists(local_path):
         try:
-            os.makedirs(os.path.dirname(local_path))
+            os.makedirs(dir_name)
         except OSError:
             # potentially, the directory has been created already
             pass
```

Comment:
This try / except construct can be simplified to `os.makedirs(dir_name, exist_ok=True)` in Python 3. This code was probably written at a time we still supported Python 2 which is a long time ago.